### PR TITLE
Initial commit for Tag-Repos job

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Pipeline file to tag OpenJ9 and OpenJ9-OMR
- Fail if Tag already exists
- Fail if tag point does not exist
- Inputs: Tag Name, Tag Annotation (Optional), OpenJ9 SHA, OMR SHA

[skip ci]

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
[Job](https://ci.eclipse.org/openj9/job/Tag-Repos)

File: buildenv/jenkins/jobs/infrastructure/tagRepos [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+// Assumes OPENJ9_SHA, OMR_SHA, and TAG_NAME are passed in Jenkins parameters
+// TAG_ANNOTATION is optional
+
+HTTP = 'https://'
+OMR_REPO = 'github.com/AdamBrousseau/openj9-omr.git'
+OPENJ9_REPO = 'github.com/AdamBrousseau/openj9.git'
```
Comment (AdamBrousseau) [inline buildenv/jenkins/jobs/infrastructure/tagRepos]: Will update this to `eclipse` once done testing

File: buildenv/jenkins/jobs/infrastructure/tagRepos [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            error MESSAGE
+        }
+
+        SHA = sh (
+                script: "git rev-parse --short ${TAG_POINT}",
+                returnStdout: true
+            )
+        echo "SHA: ${SHA}"
+        // This 'if' condition is WIP
+        if ((TAG_POINT ==~ /.*SHA.*/) || (SHA ==~ /.*TAG_POINT.*/)) {
```
Comment (AdamBrousseau) [inline buildenv/jenkins/jobs/infrastructure/tagRepos]: This is the only WIP left for me

**Comment by AdamBrousseau** [comment]
Test cases
- No Tag Name: https://ci.eclipse.org/openj9/job/Tag-Repos/2
- Default Branches: https://ci.eclipse.org/openj9/job/Tag-Repos/21/
- SHAS & Annotation: https://ci.eclipse.org/openj9/job/Tag-Repos/22/
- 1 SHA, 1 Branch, no Annotation: https://ci.eclipse.org/openj9/job/Tag-Repos/23/

Ready for review @pshipton

**Comment by AdamBrousseau** [comment]
@DanHeidinga do we want to make the Tag Annotation mandatory? Do we have a set format for the annotation? The code would be slightly less complicated if it was mandatory. I'm not sure if the annotation makes a difference from a git checkout perspective.

**Comment by keithc-ca** [comment]
The command we're using (`git describe --exact-match`) will only work with annotated tags (i.e those created by using `git tag -a ...`).

**Comment by DanHeidinga** [comment]
> @DanHeidinga do we want to make the Tag Annotation mandatory?

Yes, the release.md file says to use annotated tags as GITHUB treats them differently than simple tags.  The format should be free form text.

See https://github.com/eclipse/openj9/issues/1265 for examples of the annotation I've used for the 0.8.0 release.

**Comment by AdamBrousseau** [comment]
Updated to make ANNOTATION mandatory.
Tests
- no annotaion provided: https://ci.eclipse.org/openj9/job/Tag-Repos/30
- 1 sha, 1 branch: https://ci.eclipse.org/openj9/job/Tag-Repos/29
- only tag name and annotaiton (default branches): https://ci.eclipse.org/openj9/job/Tag-Repos/31

File: buildenv/jenkins/jobs/infrastructure/tagRepos [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+// Assumes OPENJ9_SHA, OMR_SHA, and TAG_NAME are passed in Jenkins parameters
+// TAG_ANNOTATION is optional
```
Comment (keithc-ca) [inline buildenv/jenkins/jobs/infrastructure/tagRepos]: This isn't true anymore.

Comment (AdamBrousseau) [reply buildenv/jenkins/jobs/infrastructure/tagRepos]: updated

File: buildenv/jenkins/jobs/infrastructure/tagRepos [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        } catch(e) {
+            def MESSAGE = "Tag Point: '${TAG_POINT}' does not exist in ${REPO_NAME}, stopping build."
+            echo MESSAGE
+            error MESSAGE
+        }
+
+        def SHA = sh (
+                script: "git rev-parse --short ${POINT_TYPE_PREFIX}${TAG_POINT}",
+                returnStdout: true
+            )
```
Comment (keithc-ca) [inline buildenv/jenkins/jobs/infrastructure/tagRepos]: Nit: can you adjust the indentation to be consistent with the rest of this file?

File: buildenv/jenkins/jobs/infrastructure/tagRepos [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        } catch(e) {
+            echo STATUS
+        }
+
+        def REPO_NAME = REPO.substring(REPO.lastIndexOf('/') +1, REPO.lastIndexOf('.git'))
+        echo "REPO_NAME: ${REPO_NAME}"
+
+        if (STATUS == 128) {
+            echo "Good, tag does not exist yet"
```
Comment (keithc-ca) [inline buildenv/jenkins/jobs/infrastructure/tagRepos]: Where is it documented that `128` is returned in this situation? Do we need to be specific here?

Comment (AdamBrousseau) [reply buildenv/jenkins/jobs/infrastructure/tagRepos]: I can't find it but I see 128 is used for other failure types too. So maybe we should just remove it and take any failure as "tag does not exist"

**Comment by AdamBrousseau** [comment]
updated

**Comment by AdamBrousseau** [comment]
Simplified script. Removed the two checks for existing tag and tag_point since having additional checks does not add any value. The git tag command will fail and stderr will indicate what is wrong.

**Pull request merged**